generator/views/client_counts_view.py

```diff
@@ -0,0 +1,132 @@
+            "sql": """
+              DATE_ADD(
+                {% if client_counts.first_seen_date._is_selected %}
+                  DATE_ADD(DATE(${client_counts.first_seen_date}), INTERVAL 1 DAY)
+                {% elsif client_counts.first_seen_week._is_selected %}
+                  DATE_ADD(DATE(${client_counts.first_seen_week}), INTERVAL 1 WEEK)
+                {% elsif client_counts.first_seen_month._is_selected %}
+                  DATE_ADD(PARSE_DATE('%Y-%m', ${client_counts.first_seen_month}), INTERVAL 1 MONTH)
+                {% elsif client_counts.first_seen_year._is_selected %}
+                  DATE_ADD(DATE(${client_counts.first_seen_year}, 1, 1), INTERVAL 1 YEAR)
+                {% endif %}
+                ,
+                {% if client_counts.days_since_first_seen._is_selected %}
+                  INTERVAL ${client_counts.days_since_first_seen} DAY
+                {% elsif client_counts.weeks_since_first_seen._is_selected %}
+                  INTERVAL ${client_counts.weeks_since_first_seen} WEEK
+                {% elsif client_counts.months_since_first_seen._is_selected %}
+                  INTERVAL ${client_counts.months_since_first_seen} MONTH
+                {% elsif client_counts.years_since_first_seen._is_selected %}
+                  INTERVAL ${client_counts.months_since_first_seen} YEAR
+                {% endif %}
+              ) < current_date
+              """,
+        }
+    ]
+
+    default_measures: List[Dict[str, Union[str, List[Dict[str, str]]]]] = [
+        {
+            "name": "client_count",
+            "type": "number",
+            "description": "The number of clients, "
+            "determined by whether they sent a baseline ping on the day in question.",
+            "sql": """
+              {% if client_counts.submission_date._is_selected or client_counts.days_since_first_seen._is_selected %}
+                -- This query is grouping on a dimension known to have 1 row per-client
+                COUNT(*)
+              {% else %}
+                -- This query is not grouping on a dimension that is known to have 1 row per-client
+                COUNT(DISTINCT client_id)
+              {% endif %}
+            """,
+        }
+    ]
+
+    def __init__(self, namespace: str, tables: List[Dict[str, str]]):
+        """Get an instance of a ClientCountsView."""
+        super().__init__(namespace, "client_counts", ClientCountsView.type, tables)
+
+    @classmethod
+    def from_db_views(
+        klass,
+        namespace: str,
+        is_glean: bool,
+        channels: List[Dict[str, str]],
+        db_views: dict,
+    ) -> Iterator[ClientCountsView]:
+        """Get Client Count Views from db views and app variants."""
+        dataset = next(
+            (channel for channel in channels if channel.get("channel") == "release"),
+            channels[0],
+        )["dataset"]
```

Comment:
Are we sure `channels` always has at least one element?

Reply:
It does, this comes from the [`variants`](https://github.com/mozilla/lookml-generator/blob/main/generator/namespaces.py#L62), so if there were no channels the app wouldn't exist. For example, Burnham has no "channels" per-se, but with a single variant it ends up as just `[{"channel": None, "dataset": "burnham"}]`.

This is probably not super clear from the code structure, so we should consider cleaning it up (but this problem exist in `probe-info-service` as well).

Reply:
May be worth a quick comment then.